src/utils/mailchimp.ts

```diff
@@ -0,0 +1,32 @@
+import { mailChimpUrl } from "@/components/Footer";
+import jsonp from "jsonp";
+
+const formatMailchimpUrl = (url: string) => url.replace("/post?", "/post-json?");
+
+export async function mailchimpSignup(email: string) {
+  const params = new URLSearchParams({
+    EMAIL: email,
+  });
+  const URL = `${formatMailchimpUrl(mailChimpUrl)}&${params.toString()}`;
+  return fetchJsonp(URL, { param: "c" });
+}
+
+type Options = {
+  param?: string | undefined;
+  prefix?: string | undefined;
+  name?: string | undefined;
+  timeout?: number | undefined;
+};
+
+// wrap jsonp in a promise
+function fetchJsonp(url: string, opt?: Options): Promise<unknown> {
+  return new Promise((resolve, reject) => {
+    jsonp(url, opt, (err: Error | null, data?: unknown) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(data);
+      }
+    });
+  });
+}
```

Comment:
Using jsonp instead of fetch to avoid CORS errors. This is exactly what the `mailchimp/subscribe` library does under the hood.  (used in `MailChimpForm.tsx`)

Reply:
why not just use the mailchimp library?

Reply:
The library doesn't expose this logic to be used programmatically outside of react. It only exposes a form with the "subscribe" function a render prop.